File: StrategyPattern_C++/Customer.cpp
#include "Customer.h"



Customer::Customer(BankData* bankData)
{
	this->bankData = bankData;
	bankData->registerObserver(this);
}


Customer::~Customer()
{
}

void Customer::update(string msg)
{
	this->msg = msg;
	display();
}

void Customer::display()
{
	cout << "Message : " << msg << endl;
}

File: StrategyPattern_Jave/bank/Customer.java
package bank;

public class Customer implements Observer, DisplayElement
{
	private BankData bankData;
	private String msg;
	
	public Customer(BankData bankData)
	{
		this.bankData = bankData;
		bankData.register(this);
	}
	
	public void update(String msg) 
	{
		this.msg = msg;
		display();
	}

	public void display() 
	{
		System.out.println("Message : " + msg);
	}
}
